common/thorhelper/roxiehelper.cpp

```diff
@@ -1533,51 +1533,64 @@ bool CSafeSocket::readBlocktms(StringBuffer &ret, unsigned timeoutms, HttpHelper
                 pHttpHelper->setHttpMethod(HttpMethod::GET);
         }
 
+        bool sockClosed = false;
         if (pHttpHelper && pHttpHelper->isHttp())
         {
 #define MAX_HTTP_HEADERSIZE 16000 //arbitrary per line limit, most web servers are lower, but REST queries can be complex..
             char header[MAX_HTTP_HEADERSIZE + 1]; // allow room for \0
-            sock->readtms(header, 1, MAX_HTTP_HEADERSIZE, bytesRead, timeoutms);
-            header[bytesRead] = 0;
-            char *payload = strstr(header, "\r\n\r\n");
-            if (payload)
+            const char *val = nullptr;
+            unsigned readtimeout;
+            if (timeoutms == INFINITE)
+                readtimeout = 60000;
+            else
+                readtimeout = timeoutms;
+            CTimeMon tm(readtimeout);
+            unsigned totalRead = 0;
+            unsigned remaining;
+            len = 0;
+            while (!sockClosed && !tm.timedout(&remaining))
```

Comment:
Before and if times out in readTmsAllowClose below it will fire a JSock timeout exception.
But if it times out here, it will fall through and end up firing a misleading "Badly formed HTTP header" exception.

Probably no real need for it to have this condition in the while loop, let readtmsAllowClose to hit the timeout? or explicitly fire a timeout exception?

Reply:
ok, I could check if sockClosed or timeout after loop.